File: footnest_app/lib/models/competition/competition.dart
import '../competitionseason/competition_season.dart';

class Competition {

  final int id;
  final String name;
  final String? logoPath;
  final String type;
  final List<CompetitionSeason> seasons;


  const Competition({
    required this.id,
    required this.name,
    this.logoPath,
    required this.type,
    required this.seasons,
  });


  factory Competition.fromJson(Map<String, dynamic> json) {

    return Competition(
      id: json['id'],
      name: json['name'] ?? '',
      logoPath: json['logoPath'],
      type: json['type'] ?? '',
      seasons: (json['seasons'] as List?)
          ?.map(
              (e) => CompetitionSeason.fromJson(e)
          )
          .toList()
          ?? [],
    );

  }

}

File: footnest-backend/src/main/java/com/footnest/footnest_backend/entity/Competition.java
package com.footnest.footnest_backend.entity;

import java.util.ArrayList;
import java.util.List;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "competitions")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Competition {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    private String logoPath;


    @Enumerated(EnumType.STRING)
    private CompetitionType type;


    @OneToMany(mappedBy = "competition")
    private List<CompetitionSeason> seasons = new ArrayList<>();


    public enum CompetitionType {
        LEAGUE,
        CUP
    }
}